inference-engine/src/legacy_api/src/transformations/convert_opset1_to_legacy/convert_nms_5_to_legacy.cpp

```diff
@@ -7,49 +7,54 @@
-    auto iou_threshold = ngraph::opset4::Constant::create(element::f32, Shape{}, {0.75});
-    auto score_threshold = ngraph::opset4::Constant::create(element::f32, Shape{}, {0.7});
-    auto nms = std::make_shared<ngraph::opset4::NonMaxSuppression>(boxes, scores, max_output_boxes_per_class,
-                                                                   iou_threshold, score_threshold);
+    auto max_output_boxes_per_class = ngraph::opset5::Constant::create(element::i64, Shape{}, {10});
+    auto iou_threshold = ngraph::opset5::Constant::create(element::f32, Shape{}, {0.75});
+    auto score_threshold = ngraph::opset5::Constant::create(element::f32, Shape{}, {0.7});
+    auto soft_nms_sigma = ngraph::opset5::Constant::create(element::f32, Shape{}, {0.25});
+    auto nms = std::make_shared<ngraph::opset5::NonMaxSuppression>(boxes, scores, max_output_boxes_per_class,
+                                                                   iou_threshold, score_threshold, soft_nms_sigma);
 
     ngraph::matcher_pass_callback callback = [](pattern::Matcher &m) {
-        auto nms_4 = std::dynamic_pointer_cast<ngraph::opset4::NonMaxSuppression>(m.get_match_root());
-        if (!nms_4) {
+        auto nms_5 = std::dynamic_pointer_cast<ngraph::opset5::NonMaxSuppression>(m.get_match_root());
+        if (!nms_5) {
             return false;
         }
 
-        const auto new_args = nms_4->input_values();
-        const auto& arg2 = new_args.size() > 2 ? new_args.at(2) : ngraph::opset4::Constant::create(element::i32, Shape{}, {0});
-        const auto& arg3 = new_args.size() > 3 ? new_args.at(3) : ngraph::opset4::Constant::create(element::f32, Shape{}, {.0f});
-        const auto& arg4 = new_args.size() > 4 ? new_args.at(4) : ngraph::opset4::Constant::create(element::f32, Shape{}, {.0f});
+        const auto new_args = nms_5->input_values();
+        const auto& arg2 = new_args.size() > 2 ? new_args.at(2) : ngraph::opset5::Constant::create(element::i32, Shape{}, {0});
+        const auto& arg3 = new_args.size() > 3 ? new_args.at(3) : ngraph::opset5::Constant::create(element::f32, Shape{}, {.0f});
+        const auto& arg4 = new_args.size() > 4 ? new_args.at(4) : ngraph::opset5::Constant::create(element::f32, Shape{}, {.0f});
+        const auto& arg5 = new_args.size() > 5 ? new_args.at(5) : ngraph::opset5::Constant::create(element::f32, Shape{}, {.0f});
 
         const auto max_output_boxes_per_class_rank = arg2.get_partial_shape().rank();
         const auto iou_threshold_rank = arg3.get_partial_shape().rank();
         const auto score_threshold_rank = arg4.get_partial_shape().rank();
+        const auto soft_nms_sigma_rank = arg5.get_partial_shape().rank();
 
         // Check that required ranks are not dynamic
         if (max_output_boxes_per_class_rank.is_dynamic() ||
             iou_threshold_rank.is_dynamic() ||
-            score_threshold_rank.is_dynamic()) {
+            score_threshold_rank.is_dynamic() ||
```

Comment:
Let's remove all these checks for `is_dynamic` for these values. And instead replace the Unsqueeze(axis=0) below with `Resize([1])`. This transformation does not depend on the shape of the input value shape.

Reply:
Maybe `Reshape([1])`?

Reply:
Done.